### DISCUSSION TITLE
xarray_leaflet?

### DISCUSSION THREAD

**Discussion by lopezvoliver** [opengeos/leafmap#676] [Q&A] [answered]
Looking at the [load_raster](https://github.com/opengeos/leafmap/blob/master/examples/notebooks/05_load_raster.ipynb) notebook, I initially thought that `add_raster` worked using [xarray_leaflet](https://github.com/xarray-contrib/xarray_leaflet), but it actually uses [localtileserver](https://github.com/banesullivan/localtileserver/). 

Fortunately, I found out the newest releases of leafmap introduced a way to actually use xarray objects as shown in [this notebook](https://github.com/opengeos/leafmap/blob/master/examples/notebooks/89_image_array_viz.ipynb). I was wondering if it's feasible to include the `array_to_image` step within `add_raster`, so that `add_raster` may accept xarray objects directly? 

I already tried using chunked xarray objects and it works nicely as well, but I was also wondering whether it's possible to leverage a dask client to generate/fetch the tiles faster? I think this is one of the original goals of `xarray_leaflet`?

**Comment by giswqs** [comment]
Good suggestions! PRs are welcome. Adding xarray support for `add_raster` should be easy. I have not used use dask extensively. Not quite sure about this part.

**Comment by giswqs** [comment, marked as answer]
xarray supports has been added in #678
